javascript/implementation/Database.ts

```diff
@@ -197,6 +200,81 @@ export class Database {
         return this.lastLinkToExtend;
     }
 
+    /**
+     * Reset the properties associated with a container to a previous time.
+     * @param toTime optional timestamp to reset to. If not provided, the properties will be deleted.
+     * @param bundlerOrComment optional bundler to add this change to, or a string to add a comment to a new bundle.
+     */
+    public async resetContainerProperties(
```

Comment:
I wanted this to be part of the Container base class but I ran into issues with constructing a container due to circular imports. I think its still reasonable for it to exist in the database though.